lib/network/modules/components/Node.js

```diff
@@ -258,8 +267,20 @@ class Node {
     util.selectiveNotDeepExtend(fields, parentOptions, newOptions, allowDeletion);
 
     Node.checkMass(newOptions);
-    if (globalOptions.opacity !== undefined && newOptions.opacity === undefined) {
-      newOptions.opacity = globalOptions.opacity;
+
+    
+    if (parentOptions.opacity !== undefined) {
+      if (Number.isNaN(parentOptions.opacity) || !Number.isFinite(parentOptions.opacity) || parentOptions.opacity < 0 || parentOptions.opacity > 1) {
```

Comment:
This insanity can actually be simplified to:
```javascript
!(0 <= opacity && opacity <= 1)
```

Reply:
Created a function called `checkOpacity`